packages/@aws-cdk/cdk/lib/cloudformation/fn.ts

```diff
@@ -123,15 +124,29 @@ export class FnConcat extends FnJoin {
       } else if (isConcatIntrinsic(el)) {
         const values = concatIntrinsicValues(el);
         listOfValues.splice(i, 1, ...values);
-        i += values;
+        i += values.length;
+      } else if (i > 0
+              && typeof listOfValues[i - 1] === 'string'
+              && typeof listOfValues[i] === 'string'
+              && !unresolved(listOfValues[i - 1])
+              && !unresolved(listOfValues[i])) {
+        // Previous & current both are plain strings, concatenating them right here...
+        listOfValues[i - 1] += listOfValues[i];
+        listOfValues.splice(i, 1);
       } else {
         i++;
       }
     }
-
     super('', listOfValues);
     this.listOfValues = listOfValues;
   }
+
+  public resolve() {
+    if (this.listOfValues.length === 1) {
+      return this.listOfValues[0];
+    }
+    return FnJoin.prototype.resolve.call(this);
```

Comment:
```
return super.resolve();
```

Please. We're not in ES5 land anymore.

Reply:
TypeScript didn't let me when I tried. It lets me now, so I guess my language server must have been bloated with garbage. Duh.